client/lib/analytics/test/survicate.js

```diff
@@ -127,6 +129,26 @@ describe( 'survicate', () => {
 		} );
 	} );
 
+	describe( 'isUserOnAnonymousPaths', () => {
+		test.each( [
+			// Anonymous paths should return true
+			[ '/log-in', true ],
+			[ '/setup/onboarding/user', true ],
+			[ '/log-in/lostpassword', true ],
+			[ '/account/user-social', true ],
+			// Non-anonymous paths should return false
+			[ '/sites', false ],
+			[ '/home', false ],
+		] )( 'should return %s for path %s', ( pathname, expected ) => {
+			Object.defineProperty( window, 'location', {
```

Comment:
Just noticed we're not cleaning up those `window` and `document` changes after the tests, and that could potentially bleed through this test.

Reply:
Yeah, nice catch, [updated it here](https://github.com/Automattic/wp-calypso/pull/106225/commits/0080f164df73c2ac1b72cf0ee670ac36d3c17712#diff-1ce3cd295aca7ab519938431bafd711046bedb7bb2f7bdb0e5acf44a44d3b47aR116-R117).